modules/tinymce/src/plugins/toc/main/ts/api/Options.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Type } from '@ephox/katamari';
+
+import Editor from 'tinymce/core/api/Editor';
+import { EditorOptions } from 'tinymce/core/api/OptionTypes';
+
+const option: {
+  <K extends keyof EditorOptions>(name: K): (editor: Editor) => EditorOptions[K] | undefined;
+  <T>(name: string): (editor: Editor) => T | undefined;
+} = (name: string) => (editor: Editor) =>
+  editor.options.get(name);
+
+const register = (editor: Editor): void => {
+  const registerOption = editor.options.register;
+
+  registerOption('toc_class', {
+    processor: 'string',
+    default: 'mce-toc'
+  });
+
+  registerOption('toc_header', {
+    processor: (value) => Type.isString(value) && /^h[1-6]$/.test(value),
+    default: 'h2'
+  });
+
+  registerOption('toc_depth', {
```

Comment:
This is an actual functional change here, as this used to allow a `string` or `number`. However the `string` version was undocumented and we'd mentioned previously we wanted to move away from allowing coercion.